### ISSUE TITLE
Setup with webpack-dev-server

### ISSUE THREAD

**Issue by AdrienLemaire** [felixmosh/i18next-hmr#10]
Hi, I'm running [webpack-dev-server](https://webpack.js.org/configuration/dev-server/) locally, and couldn't get i18next-hmr to run with it.

Here's my last attempt using [devServer.before](https://webpack.js.org/configuration/dev-server/):

webpack.dev.js
```js
const { I18NextHMRPlugin } = require("i18next-hmr/plugin");
const { applyServerHMR } = require("i18next-hmr/server");
const i18n = require("./src/utilities/i18n");

const plugins = [
  new I18NextHMRPlugin({
    localesDir: path.resolve(__dirname, "src/assets/locales"),
  }),
];

const devServer = {
    contentBase: [path.join(__dirname, "dist"), path.join(__dirname, "src")],
    compress: false,
    port: 9000,
    hot: true,
    watchContentBase: true,
    https: true,
    historyApiFallback: true,
    writeToDisk: true,
    // New code to support i18n-hmr
    before: function (_app, _server, _compiler) { 
      applyServerHMR(i18n);
    },
  },

module.exports = {
  mode: "development",
  plugins,
  devServer
};
```

Is it correct that the `require('./i18n');` from your [Usage doc](https://github.com/felixmosh/i18next-hmr#usage) refers to [this i18n.js file](https://react.i18next.com/guides/quick-start#configure-i-18-next) ?
I wrote it in typescript, and webpack doesn't know how to parse it at that time. But even if reverting to js, I still get `Cannot use import statement outside a module` because of the es6 syntax.
Are the only ways to fix this issue to either write `i18n.js` as a CommonJS module, or to modify my webpack runner to support a config written in typescript or  es6 syntax?

**Comment by felixmosh** [comment]
Hi,

Thank you for reporting this issue, 🙏🏼.

`require('./i18n');` refers to project's file which creates a instance of `i18next` (via `init()`).

**Comment by AdrienLemaire** [comment]
@felixmosh thanks for the quick reply.
Indeed, I'm not using server-side rendering at the moment, got it. Looks like it's working :)

Just a couple notes:

* [Your example](https://github.com/felixmosh/i18next-hmr/blob/master/examples/react-i18next/src/i18n.js#L2) uses i18next-xhr-backend, which [appears to be deprecated](https://github.com/i18next/i18next-xhr-backend#deprecated)
* I had to create a hacky declaration file for the types. It would be nice if you could publish the `@types/i18next-hmr` package with the proper definition.

i18next-hmr.d.ts
```ts
declare module "i18next-hmr" {
  import type { i18n } from "i18next";
  export function applyClientHMR(i18n): any;
}
```

**Comment by felixmosh** [comment]
@AdrienLemaire, 
1. they deprecated it recently 
2. As of types, the lib has it's type (when you import from `i18next-hmr/client` / `i18next-hmr/server`), I will add for the index itself. 

Thank you

**Comment by felixmosh** [comment]
Types were fixed (i18next-hmr@1.5.4)

![image](https://user-images.githubusercontent.com/9304194/85036077-94e82100-b18c-11ea-9738-7fadd9bac8d8.png)

**Comment by AdrienLemaire** [comment]
Thanks @felixmosh, that's some impressive support here :heart_eyes:
